plugins/modules/aci_l3out_floating_svi_secondary_ip.py

```diff
@@ -0,0 +1,367 @@
+sent:
+  description: The actual/minimal configuration pushed to the APIC
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"], required=True),
+        l3out=dict(type="str", aliases=["l3out_name"], required=True),
+        node_profile=dict(type="str", aliases=["node_profile_name", "logical_node"], required=True),
+        interface_profile=dict(type="str", aliases=["interface_profile_name", "logical_interface"], required=True),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        pod_id=dict(type="str", required=True),
+        node_id=dict(type="str", required=True),
+        encap=dict(type="str", required=True),
+        address=dict(type="str", aliases=["addr", "ip_address"]),
+        secondary_ip=dict(type="str", aliases=["secondary_floating_address"]),
+        external_bridge_group_profile=dict(type="str"),
```

Comment:
True. Given the nature of structure of floating svi and keeping the end goal as configuration using Ansible, we should leave them there. (Let me know if there's any other workaround?)

Example 1
```
- name: Create a floating svi
   cisco.aci.aci_l3out_floating_svi: &floating_svi
      pod_id: 1
      node_id: 201
      encap: vlan-1
      address: 23.45.67.90/24
      external_bridge_group_profile: bridge1
   
  - name: Create the first floating svi secondary_ip
    cisco.aci.aci_l3out_floating_svi_secondary_ip:
      <<: *floating_svi
      secondary_ip: 25.45.67.90/24
  
  - name: Create a second floating svi secondary_ip in normal mode
    cisco.aci.aci_l3out_floating_svi_secondary_ip:
      <<: *floating_svi
      secondary_ip: 25.45.67.91/24
 
```

Example 2
```
- name: Create a floating svi
   cisco.aci.aci_l3out_floating_svi: &floating_svi
      pod_id: 1
      node_id: 201
      encap: vlan-1
      address: 23.45.67.90/24
      external_bridge_group_profile: bridge1
   
  - name: Create the first floating svi secondary_ip
    cisco.aci.aci_l3out_floating_svi_secondary_ip: &floating_svi_secondary
      pod_id: 1
      node_id: 201
      encap: vlan-1
      address: 23.45.67.90/24
      secondary_ip: 25.45.67.90/24
  
  - name: Create a second floating svi secondary_ip in normal mode
    cisco.aci.aci_l3out_floating_svi_secondary_ip:
      <<: *floating_svi_secondary
      secondary_ip: 25.45.67.91/24
```

Example 1 would be the expectation from the customer. If I exclude external_bridge_group_profile from the backend code, the user would have to exclude that parameter in Ansible configuration as well(even though they are part of the same configuration), otherwise Ansible would complain which would lead us to Example 2 which is not what we want.

Let me know if there's a workaround by just keeping Example 1 in mind?

Reply:
But if this is the use-case would it not be easier to add the secondary ip to the original module as a child?

Reply:
We can add more than one secondary ip to its parent. So we would require a list type attribute. The user would need to specify every secondary ip address again in order for deletion etc.

Reply:
Removed the un-used attributes after discussion.